connections/ethereum/connection.go

```diff
@@ -165,6 +165,34 @@ func (c *Connection) SafeEstimateGas(ctx context.Context) (*big.Int, error) {
 	}
 }
 
+func (c *Connection) SafeEstimateGasTipCap(ctx context.Context, baseFee *big.Int) (*big.Int, *big.Int, error) {
```

Comment:
Also i think this function is not only about tip cap but rather about another way of fee estimation, how about to call it EstimateGasEIP1559 or something like that